examples/synthetics_results_example_test.go

```diff
@@ -0,0 +1,422 @@
+	}
+
+	traceRespJSON, _ := json.Marshal(traceResp) // nolint:errcheck
+	fmt.Println("Got traceroute results for test (in JSON format):", string(traceRespJSON))
+	fmt.Println("Number of nodes:", len(traceResp.Nodes))
+	fmt.Println("Number of paths:", len(traceResp.Paths))
+	return nil
+}
+
+func pickNetworkMeshTestID(ctx context.Context, c *kentikapi.Client) (string, error) {
+	getAllResp, err := c.Synthetics.GetAllTests(ctx)
+	if err != nil {
+		return "", fmt.Errorf("c.Synthetics.GetAllTests: %w", err)
+	}
+
+	for _, test := range getAllResp.Tests {
+		if test.Type == synthetics.TestTypeNetworkMesh {
+			fmt.Printf("Picked network_mesh test named %q with ID %v\n", test.Name, test.ID)
+			return test.ID, nil
+		}
+	}
+	return "", errors.New("no network_mesh tests found")
+}
+
+func formatTestResultsSlice(trs []synthetics.TestResults) string {
+	var s []string
+	for _, tr := range trs {
+		s = append(s, formatTestResults(tr))
+	}
+	return fmt.Sprintf("{\n%v\n}", strings.Join(s, ", "))
+}
+
+func formatTestResults(tr synthetics.TestResults) string {
+	return fmt.Sprintf(
+		"{\n  test_id=%v\n  time=%v\n  health=%v\n  len(agents)=%v\n  agents=%v\n}",
+		tr.TestID, tr.Time, tr.Health, len(tr.Agents), formatAgentsResults(tr.Agents),
+	)
+}
+
+func formatAgentsResults(ars []synthetics.AgentResults) string {
+	var s []string
+	for _, ar := range ars {
+		s = append(s, fmt.Sprintf("{agent_id=%v health=%v len(tasks)=%v}", ar.AgentID, ar.Health, len(ar.Tasks)))
+	}
+	return fmt.Sprintf("{%v}", strings.Join(s, ", "))
+}
+
+func demonstrateSyntheticsNetworkMeshTestResults() error {
+	ctx := context.Background()
+	client, err := NewClient()
+	if err != nil {
+		return err
+	}
+
+	testID, err := pickNetworkMeshTestID(ctx, client)
+	if err != nil {
+		return err
+	}
+
+	trs, err := getLastTenTestResults(ctx, client, testID)
+	if err != nil {
+		return err
+	}
+	if len(trs) == 0 {
+		fmt.Println("No mesh test results received - exiting example")
+		return nil
+	}
+
+	getAllAgentsResp, err := client.Synthetics.GetAllAgents(ctx)
+	if err != nil {
+		return fmt.Errorf("client.Synthetics.GetAllAgents: %w", err)
+	}
+
+	m, err := newMetricsMatrix(trs, getAllAgentsResp.Agents)
+	if err != nil {
+		return fmt.Errorf("new metrics matrix: %w", err)
+	}
+
+	err = printPingLatencyMatrix(m)
+	if err != nil {
+		return fmt.Errorf("print ping latency matrix: %w", err)
+	}
+
+	err = printPingJitterMatrix(m)
+	if err != nil {
+		return fmt.Errorf("print ping jitter matrix: %w", err)
+	}
+
+	err = printPingPacketLossMatrix(m)
+	if err != nil {
+		return fmt.Errorf("print ping packet loss matrix: %w", err)
+	}
+
+	return nil
+}
+
+func getLastTenTestResults(ctx context.Context, c *kentikapi.Client, testID string) ([]synthetics.TestResults, error) {
+	trs, err := c.Synthetics.GetResultsForTests(ctx, synthetics.GetResultsForTestsRequest{
+		TestIDs:   []models.ID{testID},
+		StartTime: time.Now().Add(-time.Hour * 240), // last 10 days
```

Comment:
10 days of results is 240 samples out of which you take 10. 1 day should be enough. A better solution is to get results for the last 12 test periods (which under normal circumstances guarantees 10 samples).

Reply:
Done: 92d1019
Changed to 12 hours. Should be good enough without complicating the example